hpat/datatypes/hpat_pandas_series_functions.py

```diff
@@ -601,37 +601,107 @@ def hpat_pandas_series_isin_impl(self, values):
-    to_append : :obj:`pandas.Series` object
-               input argument
-    ignore_index:
-                 *unsupported*
-    verify_integrity:
-                     *unsupported*
+    self: :obj:`pandas.Series`
+           input series
+    to_append : :obj:`pandas.Series` object or :obj:`list` or :obj:`set`
+                Series (or list or tuple of Series) to append with self
+    ignore_index: :obj:`bool`, default False
+                If True, do not use the index labels.
+                Supported as literal value only
+    verify_integrity: :obj:`bool`, default False
+                If True, raise Exception on creating index with duplicates.
+                *unsupported*
+
     Returns
     -------
     :obj:`pandas.Series`
          returns :obj:`pandas.Series` object
+         Concatenated Series
+
     """
 
     _func_name = 'Method append().'
 
-    if not isinstance(self, SeriesType) or not isinstance(to_append, SeriesType):
+    if not isinstance(self, SeriesType):
+        raise TypingError(
+            '{} The object must be a pandas.series. Given self: {}'.format(_func_name, self))
+
+    if not (isinstance(to_append, SeriesType)
+            or (isinstance(to_append, (types.UniTuple, types.List)) and isinstance(to_append.dtype, SeriesType))):
+        raise TypingError(
+            '{} The argument must be a pandas.series or list/tuple of pandas.series. \
+            Given to_append: {}'.format(_func_name, to_append))
+
+    # currently we will always raise this in the end, i.e. if no impl was found
+    # TODO: find a way to stop compilation early and not proceed with unliteral step
+    if not (isinstance(ignore_index, types.Literal) and isinstance(ignore_index, types.Boolean)
+            or isinstance(ignore_index, types.Omitted)
+            or ignore_index is False):
+        raise TypingError(
+            '{} The ignore_index must be a literal Boolean constant. Given: {}'.format(_func_name, ignore_index))
+
+    if not (verify_integrity is False or isinstance(verify_integrity, types.Omitted)):
         raise TypingError(
-            '{} The object must be a pandas.series. Given self: {}, to_append: {}'.format(_func_name, self, to_append))
+            '{} Unsupported parameters. Given verify_integrity: {}'.format(_func_name, verify_integrity))
+
+    # ignore_index value has to be known at compile time to select between implementations with different signatures
+    if ((isinstance(ignore_index, types.Literal) and ignore_index.literal_value is True)
+            or (isinstance(ignore_index, bool) and ignore_index is True)):
+        # implementations that ignore series index
+        if isinstance(to_append, SeriesType):
+            def hpat_pandas_series_append_single_impl(self, to_append, ignore_index=False, verify_integrity=False):
+
+                new_data = hpat.hiframes.api._append(self._data, to_append._data)
+                new_index = numpy.arange(len(self._data) + len(to_append._data))
```

Comment:
Shouldn't it be ```None```?

Reply:
Yeap, right, not needed. Will fix in next commit.